style: enhance and reformat issue template

kehugter opened the pull request:
Add missing required information for new issues
Redo Markdown format for a cleaner raw display
Standarise requirements/terms with CONTRIBUTING.md

Template standard needed to finish #2983.

What do you think about moving `ISSUE_TEMPLATE.md` out of the root, into GH's custom `.github` dir?

Commit:
style: enhance and reformat issue template

Add missing required information for new issues
Redo Markdown format for a cleaner raw display
Standarise requirements/terms with CONTRIBUTING.md

zetok commented:
> What do you think about moving `ISSUE_TEMPLATE.md` out of the root, into GH's custom `.github` dir?

I don't see any reason to not do that.

As for the PR, I'm not sure if changing `**` to `####` is a good thing, since with this font will be overly large, while currently it's IMHO sufficiently visually marked. Perhaps use 5 `#`, which will give same size?

Commit:
style: edit issue template, move it to .github dir

Reduce the titles size in the issue template.
Move the issue template file to GitHub's custom `.github` directory.
